waqi_client_async/__init__.py

```diff
@@ -29,30 +33,51 @@ class OverQuota(APIError):
     pass
 
 
+class UnknownCity(APIError):
+    """Raised when the provided city could not be found."""
+
+    pass
+
+
+class UnknownID(APIError):
+    """Raised when the provided ID could not be found."""
+
+    pass
+
+
 class UnknownStation(APIError):
     """Raised when the provided station could not be found."""
 
     pass
 
 
 def assert_valid(result: dict) -> None:
-    if result.get("status") != "error":
-        return
+    LOGGER.debug("JSON Data: %s", result)
+    if result.get("status") == "ok":
+        if data := result.get("data") is not None:
```

Comment:
This `:=` is a Python 3.8 feature. This project was configured for Python >=3.7 (see [here](https://github.com/sbach/waqi-client-async/blob/main/pyproject.toml#L15))

Reply:
The question becomes: what is our intent here?

Reply:
HA is now on python 3.11, so I'd propose raising the minimalsupported version.

Reply:
And implemented.